test/testdrive/kafka-avro-debezium-transaction.td

```diff
@@ -0,0 +1,210 @@
+            }
+          ],
+          "connect.name": "io.debezium.connector.mysql.Source"
+        }
+      },
+      {
+        "name": "transaction",
+        "type": {
+          "type": "record",
+          "name": "Transaction",
+          "namespace": "whatever",
+          "fields": [
+            {
+              "name": "total_order",
+              "type": ["long", "null"]
+            },
+            {
+              "name": "id",
+              "type": "string"
+            }
+          ]
+        }
+      }
+    ]
+  }
+
+$ set txschema={
+    "type": "record",
+    "name": "TransactionMetadataValue",
+    "namespace": "io.debezium.connector.common",
+    "fields": [
+      {"name": "status", "type": "string"},
+      {"name": "id", "type": "string"},
+      {
+        "name": "event_count",
+        "type": ["null", "long"],
+        "default": null
+      },
+      {
+        "name": "data_collections",
+        "type": [
+          "null",
+          {
+            "type": "array",
+            "items": {
+              "type": "record",
+              "name": "ConnectDefault",
+              "namespace": "io.confluent.connect.Avro",
+              "fields": [
+                {"name": "data_collection", "type": "string"},
+                {"name": "event_count", "type": "long"}
+              ]
+            }
+          }
+        ],
+        "default": null
+      }
+    ],
+    "connect.name": "io.debezium.connector.common.TransactionMetadataValue"
+  }
+
+$ kafka-create-topic topic=data-txdata
+
+$ kafka-ingest format=avro topic=data-txdata schema=${txschema}
+{"status": "BEGIN", "id": "1", "event_count": null, "data_collections": null}
+{"status": "END", "id": "1", "event_count": {"long": 3}, "data_collections": {"array": [{"event_count": 3, "data_collection": "testdrive-data-${testdrive.seed}"}]}}
+{"status": "BEGIN", "id": "5", "event_count": null, "data_collections": null}
+{"status": "END", "id": "5", "event_count": {"long": 1}, "data_collections": {"array": [{"event_count": 1, "data_collection": "testdrive-data-${testdrive.seed}"}]}}
+
+> CREATE MATERIALIZED SOURCE data_txdata
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-data-txdata-${testdrive.seed}'
+  FORMAT AVRO USING SCHEMA '${txschema}'
+  ENVELOPE NONE 
+
+
+
+$ kafka-create-topic topic=data
+
+$ kafka-ingest format=avro topic=data schema=${schema} timestamp=1
+{"before": null, "after": {"row": {"a": 1, "b": 1}}, "source": {"file": "binlog", "pos": 0, "row": 0, "snapshot": {"boolean": false}}, "transaction": {"total_order": null, "id": "1"}}
+{"before": null, "after": {"row": {"a": 2, "b": 3}}, "source": {"file": "binlog", "pos": 1, "row": 0, "snapshot": {"boolean": false}}, "transaction": {"total_order": null, "id": "1"}}
+
+# Create a source using an inline schema.
+
+> CREATE MATERIALIZED SOURCE data_schema_inline
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-data-${testdrive.seed}'
+  WITH(tx_metadata = data_txdata)
```

Comment:
why do we point to a source instead of just the schema? then we could build the source exactly as expected?

Reply:
I'm going down this path for a couple reasons:
- I think it makes the structure more intuitive to a user
- Make more explicit some internal complexity: if we inlined the kafka info, then we'd likely make something that looks like a source but we wouldn't want to show to the user, etc
- Allow for explicit re-use of the tx topic since we do want this to be able to work cross-topic.  As you noted above, we don't currently do that, but we'll want to in the future!